packages/netsuite-adapter/src/client/graph_utils.ts

```diff
@@ -57,22 +66,31 @@ export class Graph<T> {
     })
   }
 
-  private dfs(node: GraphNode<T>, visited: Set<T[keyof T]>, resultArray: GraphNode<T>[]): void {
+  private dfs(dfsParams: DFSParameters<T>): void {
+    const { node, visited, resultArray, path, cycle } = dfsParams
     if (visited.has(node.value[this.key])) {
+      const cycleStartIndex = path?.indexOf(node)
+      if (cycleStartIndex !== -1) {
+        // node is visited & in path mean its a cycle
+        cycle?.push(...(path?.slice(cycleStartIndex) ?? []))
+      }
       return
     }
     visited.add(node.value[this.key])
+    path?.push(node)
     node.edges.forEach(dependency => {
-      this.dfs(dependency, visited, resultArray)
+      this.dfs({ node: dependency, visited, resultArray, path, cycle })
```

Comment:
I think that this way `path` can be used inside this function only, and `path.push`, `path.pop` can be removed
```suggestion
      this.dfs({ node: dependency, visited, resultArray, path: path.concat(node), cycle })
```